pkg/plugins/pg/source/source.go

```diff
@@ -166,74 +165,25 @@ func (s *Source) Read(ctx context.Context, pos record.Position) (record.Record,
 			s.snapshotter = nil
 			return record.Record{}, plugins.ErrEndData
 		}
-
 		return s.snapshotter.Next()
 	}
 
-	// TODO: Pull this code out into its own integer iterator
-	i, err := incrementPosition(pos)
-	if err != nil {
-		return record.Record{}, cerrors.Errorf("failed to increment: %w", err)
-	}
-	// query row at the incremented position i
-	query, args, err := psql.
-		Select(s.columns...).
-		From(s.table).
-		Where(sq.GtOrEq{s.key: i}).
-		Limit(1).
-		ToSql()
-	if err != nil {
-		return record.Record{}, cerrors.Errorf("failed to create read query: %w", err)
-	}
-	rows, err := s.db.QueryContext(ctx, query, args...)
-	if err != nil {
-		return record.Record{}, cerrors.Errorf("failed to query context: %w", err)
-	}
-	defer rows.Close()
-	rec := record.Record{}
-	rec, err = s.build(rows, i, rec)
-	if err != nil {
-		// if we're at the end of rows, then read from the CDCIterator
-		if cerrors.Is(err, plugins.ErrEndData) {
-			if s.buffer != nil {
-				if s.buffer.HasNext() {
-					next, err := s.buffer.Next()
-					return next, err
-				}
-			}
-			return record.Record{}, plugins.ErrEndData
+	// check cdc buffer
+	if s.cdc != nil {
+		if s.cdc.HasNext() {
+			next, err := s.cdc.Next()
+			return next, err
 		}
-		return record.Record{}, cerrors.Errorf("failed to build payload: %w", err)
 	}
-	return rec, nil
+	// TODO: implement long polling reader here
```

Comment:
For postgres users who can't uses CDC due to lack of configuration options or version issues, we can support a long polling approach to table snapshotting.

Reply:
👍